pyuvsim/telescope.py

```diff
@@ -520,6 +524,19 @@ def _obj_to_str(self, beam_model):
 
         return path
 
+    def free_shared(self):
```

Comment:
Instead of duplicating this function, it might be cleaner to do one of two things:
1) Define this as a utility function in `mpi.py` that accepts an arbitrary object. Then for both this and SkyModelData, call `mpi.free_shared(self)` at the appropriate times.
2) Define a class in mpi.py that adds the shared_win_list attribute and free_shared function, and make both BeamList and SkyModelData inherit from it.

I'd be happy to do this if you like, but this is fine if you just want to get this fixed.

Reply:
Yep, I think consolidating the code is a good idea. I think I'll take Steven's approach if I can make it work sensibly.